scripts/dotnet-cli-build/CompileTargets.cs

```diff
@@ -195,23 +185,27 @@ public static BuildTargetResult CompileStage2(BuildTargetContext c)
                 return result;
             }
 
-            // Build projects that are packed in NuGet packages, but only on Windows
-            if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+            // build projects for nuget packages
+            var packagingOutputDir = Path.Combine(Dirs.Stage2Compilation, "forPackaging");
+            Mkdirp(packagingOutputDir);
+            foreach (var project in PackageTargets.ProjectsToPack)
             {
-                var packagingOutputDir = Path.Combine(Dirs.Stage2Compilation, "forPackaging");
-                Mkdirp(packagingOutputDir);
-                foreach (var project in ProjectsToPack)
+                // Just build them, we'll pack later
+                var packBuildResult = DotNetCli.Stage1.Build(
+                    "--build-base-path",
+                    packagingOutputDir,
+                    "--configuration",
+                    configuration,
+                    Path.Combine(c.BuildContext.BuildDirectory, "src", project))
+                    .Execute();
+
+                // On Non-Windows these packages aren't published and are used for testing
+                // Net451 targets will also cause a failure, so let's ignore them for now
+                if (CurrentPlatform.IsWindows)
```

Comment:
cc @piotrpMSFT You had a better idea for this, can't remember what you had suggested.

Constraining the build framework based on OS?